src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -704,12 +704,48 @@ serialWith func (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 -- | Same as applicative '*>'. Run two folds serially one after the other
 -- discarding the result of the first.
 --
--- /Unimplemented/
---
+
+{-# ANN type SerialRunner Fuse #-}
+data SerialRunner sL sR
+    = SerialLeft !sL 
+    | SerialRight !sR
+
 {-# INLINE serial_ #-}
-serial_ :: -- Monad m =>
+serial_ :: Monad m =>
     Fold m x a -> Fold m x b -> Fold m x b
-serial_ _f1 _f2 = undefined
+serial_ (Fold stepL beginL _) (Fold stepR beginR doneR) =
+    Fold step begin done
+
+    where
+
+    begin = do
+        resL <- beginL
+        resR <- beginR
+        return
+            $ case resL of
+                  Partial sl -> Partial $ SerialLeft sl                         
+                  Done _ -> first SerialRight resR
+
+    step (SerialLeft sL) a = do
+        resL <- stepL sL a
+        case resL of
+            Partial sL1 -> return $ Partial $ SerialLeft sL1                
+            Done _ -> do
+                resR <- beginR
+                case resR of
+                    Partial sr -> return $ Partial $ SerialRight sr  
+                    Done bR -> return $ Done bR    
+
+    step (SerialRight sR) a = do
+        resR <- stepR sR a
+        case resR of
+                  Partial sR1 -> return $ Partial $ SerialRight sR1
+                  Done bR -> return $ Done bR
+
+    done (SerialLeft _) = error "Done from left fold is invalid."
```

Comment:
We can probablye eliminate this by design if we use the following constructors instead,
`SerialBoth sL sR`
`SerialLeft sL bR`
This is fine too.

Reply:
A better implementation is to be a little lazy, like the language itself.
Don't scrutinize `initialR` unless it is required.

Reply:
Need some discussion on it.